### PR TITLE
[ADD] MRP: Unbuild orders

### PR DESCRIPTION
**Pull request by hojo-odoo**
This PR adds a new doc detailing how to dismantle products using unbuild orders.

### CODE REVIEW

**Comment by hojo-odoo** [comment]
Hey @brse-odoo and @Felicious,

This doc is ready for the first round of review.

Thanks,
hojo

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            for a product with zero or fewer units on hand.
+
+After completing an unbuild order, inventory counts automatically update based on the quantity of
+products unbuilt, and the quantity of components reclaimed.
+
+.. example::
+   A `Coat Rack` product is comprised of one `Wooden Pole` component and six `Wooden Dowel`
+   components.
+
+   An unbuild order is created for one unit of the `Coat Rack`. Once the order is completed, the
```
Comment (Felicious) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:80]:
```suggestion
   When an unbuild order is created for one unit of the `Coat Rack` and completed, the
```

**Review by Felicious — APPROVED** [review]
Hi @hojo-odoo, this is a really straightforward doc, and I had almost nothing to add. You did a great job! I filled out the form and the flow works as intended. Feel free to move this onto the next stage of review, when ready! :smile:

**Comment by hojo-odoo** [comment]
Hey @ksc-odoo,

This doc is ready for final review.

Thanks,
hojo

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+.. |BoM| replace:: :abbr:`BoM (Bill of Materials)`
+
+In some cases, it is necessary to dismantle manufactured products into their individual components.
+This may be required if too many units of a product were built, or if the components of one product
+must be reclaimed to use in the manufacturing of another.
+
+In Odoo *Manufacturing*, products can be dismantled, and their components returned to inventory,
+using *unbuild orders*. By using unbuild orders to accomplish this task, inventory counts for the
+finished product and its components remain accurate, based on the quantity of products dismantled,
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:13]:
```suggestion
finished product, and its components, remain accurate, based on the quantity of products dismantled,
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+and the quantity of components reclaimed.
+
+Create unbuild order
+====================
+
+A new unbuild order can be created by navigating to :menuselection:`Manufacturing --> Operations -->
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:19]:
```suggestion
A new unbuild order can be created by navigating to :menuselection:`Manufacturing app --> Operations -->
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+Finally, once the product has been unbuilt, click the :guilabel:`Unbuild` button at the top of the
+order to confirm that it has been completed.
+
+.. image:: unbuild_orders/unbuild-order.png
+   :align: center
+   :alt: A filled-out unbuild order.
+
+.. warning::
+   While it is possible to create unbuild orders for products that have zero or fewer units on hand,
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:58]:
```suggestion
   While it is possible to create unbuild orders for products that have zero (or fewer) units on-hand,
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+   this is not advised, since it can lead to inventory inconsistencies.
+
+   If an unbuild order is created for a product with zero or fewer units on hand, a pop-up appears,
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:61]:
```suggestion
   If an unbuild order is created for a product with zero (or fewer) units on-hand, a pop-up window appears,
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+   warning the user that there is an insufficient quantity to unbuild.
+
+   To ignore the warning and proceed with the unbuild order, click :guilabel:`Confirm` at the bottom
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:64]:
```suggestion
   To ignore the warning, and proceed with the unbuild order, click :guilabel:`Confirm` at the bottom
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+   of the pop-up window. To return to the unconfirmed unbuild order, click :guilabel:`Discard`
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:65]:
```suggestion
   of the pop-up window. To return to the unconfirmed unbuild order, click :guilabel:`Discard`,
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+   instead.
+
+   .. image:: unbuild_orders/insufficient-quantity.png
+      :align: center
+      :alt: The insufficient quantity pop-up that appears after trying to confirm an unbuild order
+            for a product with zero or fewer units on hand.
+
+After completing an unbuild order, inventory counts automatically update based on the quantity of
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:73]:
```suggestion
After completing an unbuild order, inventory counts automatically update, based on the quantity of
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+   An unbuild order is created for one unit of the `Coat Rack`. Once the order is completed, the
+   on-hand quantity of `Coat Racks` decreases by one, while the on-hand quantities of `Wooden Poles`
+   and `Wooden Dowels` increase by one and six, respectively.
+
+Scrap unusable components
+=========================
+
+In some cases, components may be unusable after the unbuilding process is completed. To ensure that
+inventory counts accurately reflect the quantity of usable components on hand, any component that
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:88]:
```suggestion
inventory counts accurately reflect the quantity of usable components on-hand, any component that
```

**Review by ksc-odoo — APPROVED** [review]
Hey @hojo-odoo -- just finished my Final Review of this PR. Great job! Only had a small amount of super-minor suggestions for you to look at. Once you make the necessary adjustments, and feel like this one is ready for the next stage, feel free to tag it for Tech Review. Thanks! 👍

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

This is ready for technical review.

Thanks,
hojo

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+select it in the :guilabel:`Manufacturing Order` field.
+
+In the :guilabel:`Source Location` field, select the location where the product being unbuilt is
+currently stored.
+
+In the :guilabel:`Destination Location` field, select the location where the reclaimed components
+are stored after the unbuild order is completed.
+
+If the :guilabel:`Lots & Serial Numbers` feature is enabled in the settings of the *Inventory* app,
+a :guilabel:`Lot/Serial Number` field appears on the unbuild order, which can be used to specify the
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:43]: The text for the "Lots & Serial Numbers" feature in the settings is not present on the UI in this instruction, so no guilabel :)
```suggestion
If the *Lots & Serial Numbers* feature is enabled in the settings of the *Inventory* app, a
:guilabel:`Lot/Serial Number` field appears on the unbuild order, which can be used to specify the
```

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+Finally, once the product has been unbuilt, click the :guilabel:`Unbuild` button at the top of the
+order to confirm that it has been completed.
+
+.. image:: unbuild_orders/unbuild-order.png
+   :align: center
+   :alt: A filled-out unbuild order.
+
+.. warning::
+   While it is possible to create unbuild orders for products that have zero (or fewer) units
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:58]: Just want to double check that "zero (or fewer) units" is correct here– can there be negative units on hand? :)

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+   on-hand, this is not advised, since it can lead to inventory inconsistencies.
+
+   If an unbuild order is created for a product with zero (or fewer) units on-hand, a pop-up window
+   appears, warning the user that there is an insufficient quantity to unbuild.
+
+   To ignore the warning, and proceed with the unbuild order, click :guilabel:`Confirm` at the bottom
+   of the pop-up window. To return to the unconfirmed unbuild order, click :guilabel:`Discard`,
+   instead.
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:66]: Line over 100 chars
```suggestion
   To ignore the warning, and proceed with the unbuild order, click :guilabel:`Confirm` at the
   bottom of the pop-up window. To return to the unconfirmed unbuild order, click
   :guilabel:`Discard`, instead.
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @hojo-odoo, nice job on this brand new MRP doc! I only have a couple quick fixes and one check, please see below and tag me for another quick look! Thank you

File: content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst (hunk 11 above)
Comment (hojo-odoo) [reply content/applications/inventory_and_mrp/manufacturing/management/unbuild_orders.rst:58]: Yep, you can do it with negative units on hand.

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

I implemented both of your changes. This is ready for merge.

Thanks,
hojo

**Review by samueljlieber — APPROVED** [review]
Nice work @hojo-odoo, Approved!
.....

@robodoo r+

**Pull request closed without merging**